### PR TITLE
web: Add ~http and ~tcp filters

### PR DESCRIPTION
**Pull request by s4chin**
As in #1915

### CODE REVIEW

File: mitmproxy/tools/web/app.py [hunk 1]
```diff
@@ -45,6 +46,7 @@ def flow_to_json(flow: mitmproxy.flow.Flow) -> dict:
         f["error"] = flow.error.get_state()
 
     if isinstance(flow, http.HTTPFlow):
+        f["http"] = True
```
Comment (mhils) [inline mitmproxy/tools/web/app.py]: I think we already transmit flow.type, so we can just use that. :)

File: web/src/js/filt/filt.peg [hunk 2]
```diff
@@ -142,6 +146,10 @@ function contentType(regex){
     contentTypeFilter.desc = "content type matches " + regex;
     return contentTypeFilter;
 }
+function tcpFilter(flow){
+    return !!flow.tcp;
+}
+tcpFilter.desc = "is TCP Flow";
```
Comment (mhils) [inline web/src/js/filt/filt.peg]: "is a TCP flow"

File: mitmproxy/tools/web/app.py (hunk 1 above)
Comment (s4chin) [inline mitmproxy/tools/web/app.py]: Will do

File: web/src/js/filt/filt.peg (hunk 2 above)
Comment (mhils) [inline web/src/js/filt/filt.peg]: No worries. We do reviews for a reason. :)

**Comment by s4chin** [comment]
I did as per review.

**Comment by mhils** [comment]
Thanks!

**Pull request merged**